eventqueue/processor.go

```diff
@@ -93,17 +94,15 @@ func (p *Processor[T]) Dequeue(key string) error {
 // Close stops the processor.
 // This method blocks until the processor loop returns.
 func (p *Processor[T]) Close() error {
-	if !p.stopped.CompareAndSwap(false, true) {
-		// Already stopped
+	defer p.wg.Wait()
```

Comment:
Ok I understand now what you were saying a couple of weeks back. Ok, this makes sense.